Common/include/Luau/Common.h

```diff
@@ -1,6 +1,8 @@
 // This file is part of the Luau programming language and is licensed under MIT License; see LICENSE.txt for details
 #pragma once
 
+#define LUAU_VERSION "Luau 0.636"
```

Comment:
I don't love the idea of just having a hardcoded version string like this, particularly because it means that we'll have to manually update this string every time.

I think the "right" way to do this might be something like... resolving the version by finding the most recent git release tag at build-time, which is kind of awful in-and-of-itself. Similarly, it would be nice to include the slug for the git hash in the release string as well.

Reply:
cc @andyfriesen do you have some thoughts on a good way to do this beyond hard-coding the version string?

Reply:
If only there was a tool that we ran each time we sync code that we could.modify to handle this.